scripts/reboot_helper.py

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env python3
+#
+# reboot_helper.py
+#
+# Utility helper for reboot within SONiC
+
+import sys
+import syslog
+import sonic_platform
+
+chk_log_level = syslog.LOG_ERR
+
+
+def _log_msg(lvl, pfx, msg):
+    if lvl <= chk_log_level:
+        print("{}: {}".format(pfx, msg))
+        syslog.syslog(lvl, msg)
+
+
+def log_err(m):
+    _log_msg(syslog.LOG_ERR, "Err", m)
+
+
+def log_info(m):
+    _log_msg(syslog.LOG_INFO, "Info",  m)
+
+
+def log_debug(m):
+    _log_msg(syslog.LOG_DEBUG, "Debug", m)
+
+
+# Global variable for platform chassis
+platform_chassis = None
+
+
+def load_platform_chassis():
+    global platform_chassis
+
+    # Load new platform API class
+    try:
+        platform_chassis = sonic_platform.platform.Platform().get_chassis()
+    except Exception as e:
+        log_err("Failed to instantiate Chassis due to {}".format(repr(e)))
+        return False
+
+    if not platform_chassis:
+        log_err("Platform chassis is not loaded")
+        return False
+
+    return True
+
+
+def reboot_module(module_name):
+    """Reboot the specified module by invoking the platform API"""
+
+    # Load the platform chassis if not already loaded
+    if not load_platform_chassis():
+        log_err("Failed to load platform chassis")
+        return False
+
+    if not platform_chassis.is_smartswitch():
+        return False
+
+    # Iterate over the modules to find the one with the specified name
+    try:
+        # Use get_all_modules to retrieve all modules on the chassis
+        modules = platform_chassis.get_all_modules()
+
+        # Iterate over the modules to find the one with the specified name
+        for module in modules:
+            # Check if the module name matches the provided module_name
+            if module and module.get_name() == module_name:
+                # Reboot the module
+                log_info(f"Rebooting module {module_name}...")
+                try:
+                    module.reboot()
+                    log_info("Reboot command sent for module {module_name}")
+                    return True
+                except NotImplementedError:
+                    log_err("Reboot not implemented for module {module_name}.")
+                    return False
+                except Exception as e:  # noqa: F841
+                    log_err("An error occurred while rebooting module {module_name}: {e}")
+                    return False
+
+        # If the module with the given name is not found
+        log_err("Module {module_name} not found")
+        return False
+
+    except Exception as e:  # noqa: F841
+        log_err("Error occurred while rebooting module {module_name}: {repr(e)}")
+        return False
+
+
+def is_dpu():
+    """Check if script is running on DPU module"""
```

Comment:
This would return `True` only for NPU and not for DPUs, (to check if the current platform is a DPU platform I think we can use the same logic used in [docker_image_ctl.j2](https://github.com/sonic-net/sonic-buildimage/blob/a5a9dacad7fadd7fdd064858f2a586424ad4f766/files/build_templates/docker_image_ctl.j2#L375)

Reply:
updated. Thank you!